include/AssembleFaceElemSolverAlgorithm.h

```diff
@@ -0,0 +1,124 @@
+/*------------------------------------------------------------------------*/
+/*  Copyright 2014 Sandia Corporation.                                    */
+/*  This software is released under the license detailed                  */
+/*  in the file, LICENSE, which is located in the top-level Nalu          */
+/*  directory structure                                                   */
+/*------------------------------------------------------------------------*/
+
+
+#ifndef AssembleFaceElemSolverAlgorithm_h
+#define AssembleFaceElemSolverAlgorithm_h
+
+#include <SolverAlgorithm.h>
+#include <ElemDataRequests.h>
+#include <ScratchViews.h>
+#include <SimdInterface.h>
+#include <SharedMemData.h>
+
+namespace stk {
+namespace mesh {
+class Part;
+}
+}
+
+namespace sierra{
+namespace nalu{
+
+class Realm;
+
+class AssembleFaceElemSolverAlgorithm : public SolverAlgorithm
+{
+public:
+  AssembleFaceElemSolverAlgorithm(
```

Comment:
This class will be computed based on the known face/element pair - which will be the same for all kernels attached to it. It may be best to expect that this class provides the scratchMeStruct behind the scenes.